crates/sync/src/mutex.rs

```
@@ -0,0 +1,102 @@
+        MutexGuard { mutex: self }
+    }
+
+    pub fn try_lock(&self) -> Option<MutexGuard<'_, T, R>> {
+        if self.locked.swap(true, Ordering::Acquire) {
+            return None;
+        }
+        Some(MutexGuard { mutex: self })
+    }
+
+    // TODO: Make safe with so can only be called when in locked state.
+    pub const unsafe fn leak(&self) -> &mut T {
```

Clippy (1.96.0)

[warning] unsafe function's docs are missing a `# Safety` section

(warning)